src/js/binaryen.js-post.js

```diff
@@ -709,6 +709,25 @@ function wrapModule(module, self = {}) {
       'wait64'(ptr, expected, timeout) {
         return Module['_BinaryenAtomicWait'](module, ptr, expected, timeout, Module['i64']);
       }
+    },
+    'module'() {
+      return UTF8ToString(Module['_BinaryenMemoryImportGetModule'](module));
+    },
+    'base'() {
+      return UTF8ToString(Module['_BinaryenMemoryImportGetBase'](module));
+    },
+    'initial'() {
+      return Module['_BinaryenMemoryGetInitial'](module);
+    },
+    'shared'() {
+      return Boolean(Module['_BinaryenMemoryIsShared'](module));
+    },
+    'max'() {
+      if (Module['_BinaryenMemoryHasMax'](module)) {
+        return Module['_BinaryenMemoryGetMax'](module);
+      } else {
+        return Infinity;
+      }
```

Comment:
Is it intended that these are mixed with the instructions like `memory.fill`? Seems that it's quite easy to mistake them for instructions like `memory.shared`. Also, in light of multi memory, there'll likely need to be a memory wrapper anyhow to deal with any memory, so I wouldn't expect these to last for very long.

Reply:
Yes, see where Alon said "I think it's ok as it is." above.

I'm just making these easier to use, if this hasn't been updated for multiple memory, than someone else will need to help make those changes.

Reply:
Hmm, has the following variant been considered?

```
Module#getMemoryModule
Module#getMemoryBase
Module#getMemoryInitial
Module#getMemoryShared
Module#getMemoryMax
```

Those would about match `Module#setMemory` for now, which is similarly subject to change, with all of them not conflicting with instructions.

Reply:
I've not considered anything (maybe others have) because I'm just converting what is here from "expensive grab-bag" to "individual functions". Further improvements can be done later, as @tlively mentioned in the primary thread.

Reply:
Perhaps this is a misunderstanding. I am not suggesting any significant changes in the above variant, just to move / rename these from `memory` (which so far covers exclusively instructions) to the `Module` instance (where methods of this kind without wrappers would typically go). Basically next to `setMemory`.

Reply:
I'm going to leave that up to @kripken since he said it was fine to have these here when I asked.

Reply:
Hmm, good point @dcodeIO , you're right that this would mix instructions like `memory.fill()` with helpers. We've put the helpers on the relevant JS objects using `makeExpressionWrapper`. So I agree that adding new `getX`/`setX` functions alongside those is more consistent. That seems best.

@phated Sorry for not reading this more in detail the first time! (I am not as up to date on the JS API as I used to be, since I've been busy with other things...)

Reply:
Can someone else pick up these changes? We really need these changes in Grain and I don't have time to continue to make these large changes, as I'm actively needing to interview for jobs.